exercises/react-vite/09-making-http-requests/04-solution/src/pages/RestaurantList/RestaurantList.tsx

```diff
@@ -114,19 +85,35 @@ const RestaurantList: React.FC = () => {
           </div>
         </form>
 
-        {restaurants.data ? (
-          restaurants.data.map(({ _id, address, images, name, slug }) => (
-            <ListItem
-              key={_id}
-              address={address}
-              name={name}
-              slug={slug}
-              thumbnail={images.thumbnail}
-            />
-          ))
-        ) : (
-          <p>No restaurants.</p>
+        {city && restaurantsResponse.error && (
+          <p aria-live="polite" className="restaurant">
+            Error loading restaurants: {restaurantsResponse.error.message}
+          </p>
+        )}
+
+        {city && restaurantsResponse.isPending && (
+          <p aria-live="polite" className="restaurant loading">
+            Loading restaurants…
+          </p>
         )}
+
+        {city &&
+          restaurantsResponse.data &&
+          (restaurantsResponse.data.length === 0
+            ? !restaurantsResponse.isPending && (
+                <p aria-live="polite">No restaurants found.</p>
+              )
+            : restaurantsResponse.data.map(
+                ({ _id, slug, name, address, images }) => (
+                  <ListItem
+                    key={_id}
+                    address={address}
+                    name={name}
+                    slug={slug}
+                    thumbnail={images.thumbnail}
+                  />
+                ),
+              ))}
```

Comment:
I think this pattern above already existed before this PR, yeah? It's hard to read and I think we should make it better (in a separate ticket) 

If its all new to this PR, we should change it before merging.